### ISSUE TITLE
Is there a makeMarkdown function?

### ISSUE THREAD

**Issue by Gacnt** [showdownjs/showdown#233]
I'm trying to make a quote function to quote a forum post, which is in markup, but when clicking the quote button I'd like to the the markup and convert that forum post into Markdown and put it into the textarea, is there such a function?

**Comment by tivie** [comment]
No

**Comment by tivie** [comment]
@smoleniuch It's a feature we're developing for version 2.0

**Comment by mattkwiecien** [comment]
Is there a timeline on when 2.0 will be released?  

I have a PR ready for `showdownjs/ng-showdown` that adds `makeMarkdown` directive to the `ng-showdown` directive, but it simply needs the latest version of `showdown`.

**Comment by charlie-s** [comment]
I think I must be missing something – isn't this project described as "a bidirectional MD to HTML to MD converter"?

**Comment by smoleniuch** [comment]
@charlie-s 
I did not find any information about bidirectional flow, can you provide a sample??

Anyway the feature would be very useful.

**Comment by tivie** [comment]
@charlie-s The bidirectional is available in master and develop branches and to the soon to be released version 2.0-alpha-1

**Comment by tivie** [comment]
@smoleniuch @Gacnt @charlie-s @mattkwiecien @DougBeney  @jdempster @Blaza @matthew-inamdar @jdhankins @circleart @ryantuck

Just pushed the reverse converter to a.x branch. You can expect a 1.9.0 release with an **[EXPERIMENTAL]** reverse parser tomorrow.

> ## Donate [![Click here to lend your support to: ShowdownJS website and testing platform and make a donation at pledgie.com](https://pledgie.com/campaigns/35166.png?skin_name=chrome)](https://pledgie.com/campaigns/35166)
>As you know, ShowdownJS is a free library and it will remain free forever. However, maintaining and improving the library costs time and money.
>If you like my work and find the library useful, please donate [through Patreon](https://patreon.com/showdownjs) or directly [through paypal](https://www.paypal.me/tiviesantos)!! Your contribution will be greatly appreciated and help us continue to develop this awesome library.

**Comment by sojs-coder** [comment]
I understand that the `makeMarkdown` function _has_  been released, but it does not work in node...
`window is not defined`
Is there some way around this?

**Comment by schroeder-g** [comment]
@sojs-coder @tivie bumping this. Still encountering `window is not defined`. Any workarounds here?

![image](https://user-images.githubusercontent.com/66321261/156479441-edc91c6d-9e43-42e5-8ce2-fd2884cef17f.png)

**Comment by tivie** [comment]
Yes. The master build should be compatible with node. Uses a new dependency, JSDOM, to replace the browser's DOM Parser.

**Comment by patarapolw** [comment]
It doesn't really document how-to on Node, nor does it automatically pick up JSDOM, nor does it give relevant error message (`ReferenceError: window is not defined`).

For the quick fix, but well, I really wanted to avoid attaching to globalThis...

```js
import { JSDOM } from 'jsdom'

;(globalThis as any).window = new JSDOM('', {}).window
```

**Comment by 0xd6438c** [comment]
January 2023. "bidirectional" markdown tool still doesn't convert back to markdown, even ChatGPT is confused.

**Comment by michaelcpuckett** [comment]
This issue still seems to be present in the npm release.

**Comment by andrewlu0** [comment]
```
  const bodyMarkdown = converter.makeMarkdown(
          bodyHtml,
          new JSDOM("", {}).window.document,
        );
``` 

this works for me with JSDOM and showdown 2.1.0
